src/libraries/System.Net.Security/src/System/Net/NegotiateAuthenticationPal.Windows.cs

```diff
@@ -664,6 +664,29 @@ public override unsafe bool VerifyMIC(ReadOnlySpan<byte> message, ReadOnlySpan<b
                 }
             }
 
+            public override unsafe TReturn DeriveKeyFromSessionKey<TState, TReturn>(Func<ReadOnlySpan<byte>, TState, TReturn> keyDerivationFunction, TState state)
+            {
+                Debug.Assert(_securityContext is not null);
+
+                SecPkgContext_SessionKey sessionKey = default;
+                int result = Interop.SspiCli.QueryContextAttributesW(ref _securityContext._handle, Interop.SspiCli.ContextAttribute.SECPKG_ATTR_SESSION_KEY, &sessionKey);
```

Comment:
Some other APIs use `SSPIWrapper.QueryBlittableContextAttributes` but that doesn't seem to great with this API as:

+ It returns a bool to see if it worked or not which makes it hard to get the original error to the caller on a failure
+ The whole SafeHandle and freeing setup seems quite complicated
  + This is compounded because it seems like I need a custom handle to point to the `SessionKey` field

So in the end I just manually call the API and `FreeContextBuffer` the required address. Happy to change it if needed but I wasn't fully sure what the benefits really would be